larwirecell/Components/DepoSetSimChannelSink.cxx

```diff
@@ -0,0 +1,302 @@
+#include "DepoSetSimChannelSink.h"
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+
+#include "WireCellGen/GaussianDiffusion.h"
+#include "WireCellIface/IDepoSet.h"
+#include "WireCellUtil/NamedFactory.h"
+#include "WireCellUtil/Units.h"
+
+WIRECELL_FACTORY(wclsDepoSetSimChannelSink,
+                 wcls::DepoSetSimChannelSink,
+                 wcls::IArtEventVisitor,
+                 WireCell::IDepoSetFilter)
+
+using namespace wcls;
+using namespace WireCell;
+
+DepoSetSimChannelSink::DepoSetSimChannelSink()
+//   : m_depo(nullptr)
+{
+  m_mapSC.clear();
+}
+
+WireCell::Configuration DepoSetSimChannelSink::default_configuration() const
+{
+  Configuration cfg;
+  cfg["anodes_tn"] = Json::arrayValue;
+  cfg["anode"] = "AnodePlane"; // either `anodes_tn` or `anode`
+  cfg["rng"] = "Random";
+  cfg["artlabel"] = "simpleSC";
+  cfg["start_time"] = -1.6 * units::ms;
+  cfg["readout_time"] = 4.8 * units::ms;
+  cfg["tick"] = 0.5 * units::us;
+  cfg["nsigma"] = 3.0;
+  cfg["drift_speed"] = 1.098 * units::mm / units::us;
+  cfg["uboone_u_to_rp"] = 94 * units::mm;
+  cfg["uboone_v_to_rp"] = 97 * units::mm;
+  cfg["uboone_y_to_rp"] = 100 * units::mm;
+  cfg["u_time_offset"] = 0.0 * units::us;
+  cfg["v_time_offset"] = 0.0 * units::us;
+  cfg["y_time_offset"] = 0.0 * units::us;
+  cfg["g4_ref_time"] = -4050 * units::us; // uboone: -4050us, pdsp: -250us
+  cfg["use_energy"] = false;
+  cfg["use_extra_sigma"] = false;
+  return cfg;
+}
+
+void DepoSetSimChannelSink::configure(const WireCell::Configuration& cfg)
+{
+  m_anodes.clear();
+  auto anodes_tn = cfg["anodes_tn"];
+  for (auto anode_tn : anodes_tn) {
+    auto anode = Factory::find_tn<IAnodePlane>(anode_tn.asString());
+    m_anodes.push_back(anode);
+  }
+
+  //  Initialize a SimChannel map. Two assumptions are followed in
+  //  larsim::BackTracker::FindCimChannel()
+  //  1) fill all channels even with empty SimChannel
+  //  2) SimChannels are sorted in channel number
+  //  In DepoSetSimChannelSink::visit(), the map is reinitialized insted of
+  //  calling map::clear()
+  for (auto& anode : m_anodes) {
+    for (auto& channel : anode->channels()) {
+      // std::cout << "-> channel: " << channel << std::endl;
+      m_mapSC.emplace(channel, sim::SimChannel(channel));
```

Comment:
Suggest using `try_emplace(...)` instead of `emplace(...)`—the `try_emplace(...)` method is more efficient.